crates/wasm-encoder/src/core/code.rs

```diff
@@ -1614,8 +1614,8 @@ impl Encode for Instruction<'_> {
                 sink.push(0x18);
                 let cast_flags =
                     (from_ref_type.nullable as u8) | ((to_ref_type.nullable as u8) << 1);
-                relative_depth.encode(sink);
                 sink.push(cast_flags);
+                relative_depth.encode(sink);
```

Comment:
I'm sort of baffled how this didn't get caught by the spec test suite...

Reply:
Yeah...